### PR TITLE
feat(frontend): add modifier classes to reduce display mode cardinality

### PR DESCRIPTION
**Pull request by SOF3**
### Description

Modifiers with the same class will not be proposed in the same display mode

### Related issues

- Based on #149

### CODE REVIEW

File: pkg/frontend/tf/config/file/file.go [hunk 1]
```diff
@@ -163,16 +186,28 @@ func (p *FileProvider) loadJsonBytes(jsonBytes []byte) error {
...
 		for _, config := range p.configs {
-			newConfig := config.Clone()
-			modifier(newConfig)
-			newEntries = append(newEntries, newConfig)
+			newConfig := config.config.Clone()
+			modifier.fn(newConfig)
+
+			if config.modifierClasses.Has(modifier.class) {
+				// incompatible combination
+				continue
```
Comment (xuqingyun) [inline pkg/frontend/tf/config/file/file.go:201]: should modify old config for the same class?  concatenate display name or append extension

File: pkg/frontend/tf/config/file/file.go [hunk 2]
```diff
@@ -163,16 +186,28 @@ func (p *FileProvider) loadJsonBytes(jsonBytes []byte) error {
...
 
 	for _, modifier := range modifiers {
-		var newEntries []*tfconfig.Config
+		var newEntries []registeredConfig
 
 		for _, config := range p.configs {
-			newConfig := config.Clone()
-			modifier(newConfig)
-			newEntries = append(newEntries, newConfig)
+			newConfig := config.config.Clone()
```
Comment (xuqingyun) [inline pkg/frontend/tf/config/file/file.go:196]: why didn't it clone extensions in Clone():
https://github.com/SOF3/kelemetry/blob/2d2a0b6ca97233ae9e379651ec8808bc079b94e4/pkg/frontend/tf/config/config.go#L60

File: hack/tfconfig.yaml [hunk 3]
```diff
@@ -78,25 +81,26 @@ modifiers:
...
-#          pod: "{{.namespace}}/{{.name}}"
-#          start: "{{unixMicro .start}}"
-#          end: "{{unixMicro .end}}"
-#          limit: "100"
-#        urlTemplate: "http://test-domain/api/traces"
-#        forObject: true
-#      maxConcurrency: 100
-#      totalTimeout: 60s
+#   "00000002":
+#      priority: 20
```
Comment (xuqingyun) [inline hack/tfconfig.yaml:85]: one more space than next line

File: pkg/frontend/tf/config/file/file.go (hunk 1 above)
Comment (SOF3) [reply pkg/frontend/tf/config/file/file.go:201]: No need, the old config is included in another iteration of this loop

File: pkg/frontend/tf/config/file/file.go (hunk 2 above)
Comment (SOF3) [reply pkg/frontend/tf/config/file/file.go:196]: What do you mean by "extensions"? You mean clone the `modifier` value?

File: pkg/frontend/tf/config/file/file.go (hunk 1 above)
Comment (xuqingyun) [reply pkg/frontend/tf/config/file/file.go:201]: i don't understand,  whatever iteration when config1[modifier1] changed to config1[modifier1][modifier2], it modified a clone new config, will be skipped in this line, no changes will be saved

File: pkg/frontend/tf/config/file/file.go (hunk 2 above)
Comment (SOF3) [reply pkg/frontend/tf/config/file/file.go:196]: extensions are added to the cloned config the line below, by running modifiers on it

**Review by xuqingyun — APPROVED** [review]

File: pkg/frontend/tf/config/file/file.go (hunk 2 above)
Comment (SOF3) [reply pkg/frontend/tf/config/file/file.go:196]: same class = 同性相斥

**Pull request merged**